packages/test-kit/src/async-base-fs-contract.ts

```diff
@@ -406,5 +406,98 @@ export function asyncBaseFsContract(testProvider: () => Promise<ITestInput<IBase
                 await expect((await fs.stat(upperCaseFilePath)).isFile()).to.equal(true)
             }
         })
+
+        describe('copying files/directories', () => {
+            const SOURCE_FILE_NAME = 'file.txt'
+            const COPYFILE_EXCL = 1
+            let targetDirectoryPath: string
+            let sourceFilePath: string
+
+            beforeEach(async () => {
+                const { fs, tempDirectoryPath } = testInput
+                const { join } = fs.path
+                targetDirectoryPath = join(tempDirectoryPath, 'dir')
+                await fs.mkdir(targetDirectoryPath)
+                sourceFilePath = join(tempDirectoryPath, SOURCE_FILE_NAME)
+                await fs.writeFile(sourceFilePath, SAMPLE_CONTENT)
+            })
+
+            it('can copy file', async () => {
+                const { fs } = testInput
+                const targetPath = fs.path.join(targetDirectoryPath, SOURCE_FILE_NAME)
+                await fs.copyFile(sourceFilePath, targetPath)
+                expect(await fs.readFile(targetPath)).to.be.eql(SAMPLE_CONTENT)
+            })
+
+            it('can copy directory (without contents)', async () => {
```

Comment:
is this really an expected behavior?

(going back to above comment about "native node fs throws EISDIR when trying to copy a directory.")